checks/ldap.go

```diff
@@ -35,20 +35,26 @@ func (c *LdapChecker) Check(ctx *context.Context, extConfig external.Check) pkg.
 	check := extConfig.(v1.LDAPCheck)
 	result := pkg.Success(check, ctx.Canary)
 	var results pkg.Results
+	var err error
 	results = append(results, result)
-	ld, err := ldap.DialURL(check.Host, ldap.DialWithTLSConfig(&tls.Config{
-		InsecureSkipVerify: check.SkipTLSVerify,
-	}))
-	if err != nil {
-		return results.Failf("Failed to connect %v", err)
+
+	if ok, err := check.HydrateConnection(ctx); err != nil {
```

Comment:
Can we make this follow the ctx.HydrateConnection pattern

Reply:
`check.HydrateConnection` is internally doing that [here](https://github.com/flanksource/canary-checker/pull/959/files/217daf66778c68ca87905f91857d095a216ecba4#diff-b2b2311aec83c084649791c6470895a8e34180b9ee76afbae9d9efdfb598303cR549)